src/early-types.lisp

```diff
@@ -53,17 +53,67 @@
 (defvar *type-parsers* (make-hash-table :test 'equal)
   "Hash table of defined type parsers.")
 
+(define-condition cffi-error (error)
+  ())
+
+(define-condition type-related-error (cffi-error)
```

Comment:
This is was long overdue. `foreign-type-error` seems more inline with other CFFI names, maybe? Naming is hard...